### PR TITLE
Feature/ab2d 4032 sonarcude bugs

### PR DESCRIPTION
**Pull request by sb-dkniesteadt**
**JIRA Tickets:**

[AB2D-4032](https://jira.cms.gov/browse/AB2D-4032) - Page Enrollment into Memory

***Related Tickets***
 
### What Does This PR Do?
Clean up the code so that it passes and meets sonarcube standards

### What Should Reviewers Watch For?
If any changes could break existing functionality (threading, testing, ect)
Many of the changes might seem unnecessary at first glance,  but its also to meet sonarcube standards. 

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [x] Code checked for PHI/PII exposure

### CODE REVIEW

File: common/src/main/java/gov/cms/ab2d/common/config/Mapping.java [hunk 1]
```diff
@@ -50,8 +51,11 @@ public void init() {
                                 context.getSource().getAttestedOn().toString());
         Converter<ContractDTO, Contract> sponsorDTOSponsorConverter = new AbstractConverter<>() {
             protected Contract convert(ContractDTO source) {
+                Optional<Contract> contract = contractService.getContractByContractNumber(source.getContractNumber());
                 //noinspection OptionalGetWithoutIsPresent
-                return contractService.getContractByContractNumber(source.getContractNumber()).get();
+                if (contract.isPresent())
+                    return contract.get();
+                return null;
```
Comment (nflearl) [inline common/src/main/java/gov/cms/ab2d/common/config/Mapping.java:58]: We have already flagged this with /noinspection.  The code is structured as intended.  Put a line in so the SonarCube ignores this line.  No reason to complicated the code for a case that never will happen.

File: fhir/src/main/java/gov/cms/ab2d/fhir/Versions.java [hunk 2]
```diff
@@ -227,7 +233,10 @@ static Object instantiateEnum(FhirVersion version, String cName, String value) {
             String topClassName = version.getClassName(cName);
             Class clazz = Class.forName(topClassName);
             Method valueOf = getMethod(clazz, "valueOf", String.class);
-            return valueOf.invoke(null, value);
+            if (valueOf != null)
```
Comment (nflearl) [inline fhir/src/main/java/gov/cms/ab2d/fhir/Versions.java:236]: just assert it

File: worker/src/main/java/gov/cms/ab2d/worker/processor/StreamHelperImpl.java [hunk 3]
```diff
@@ -64,7 +65,7 @@ public String getSuffix() {
     private final long totalBytesAllowed;
 
     // The current output stream
-    protected volatile OutputStream currentStream;
+    protected final ThreadLocal<OutputStream> currentStream = new ThreadLocal<>();
```
[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/StreamHelperImpl.java:68]: I would think an AtomicReference would be a better replacement.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/ProgressTrackerIntegrationTest.java [hunk 4]
```diff
@@ -1,8 +1,13 @@
 package gov.cms.ab2d.worker.processor;
 
 import gov.cms.ab2d.bfd.client.BFDClient;
-import gov.cms.ab2d.common.model.*;
-import gov.cms.ab2d.common.repository.*;
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.Job;
```
[the remark this answers is not in the document]
Comment (sb-dkniesteadt) [reply worker/src/test/java/gov/cms/ab2d/worker/processor/ProgressTrackerIntegrationTest.java:5]: This is about AtomicReference. The only problem with it is that it doesn't have a remove/ nullifying the reference unlike threadlocal.

**Pull request merged**